### ISSUE TITLE
[BUG][redis-cli] faulty commands mess subsequent reponse

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ashish21590/redis#294]
```
What steps will reproduce the problem?
1. issue a faulty command in redis-cli
2. issue a right command -> get back a wrong reponse
3. issue a right command again -> get back the right reponse

Example session in redis 1.2.6:

redis> get a
1
redis> append a fu
(error) ERR unknown command 'append'
redis> get a
(error) ERR unknown command 'fu'
redis> get a
1

Example session in redis 1.3.16

redis> get a
"a"
redis> multi
OK
redis> incr a
QUEUED
redis> get b
QUEUED
redis> exec
1. (error) ERR value is not an integer
redis> get a
"b"
redis> get a
"a"

What is the expected output? What do you see instead?

the right response the first time you try it.

What version of the product are you using? On what operating system?

1.2.6 & 1.3.16

Please provide any additional information below.

I'm not entirely shure if this is an issue with the cli or redis itself. I'm on 
OSX snowleopard (in case this matters).
```

Original issue reported on code.google.com by `niko.dit...@googlemail.com` on 25 Jul 2010 at 9:27

**Comment by GoogleCodeExporter** [comment]
```
Hello, this is fixed in both Redis 2.0.0 RC3 and Redis master.

Cheers,
Salvatore
```

Original comment by `anti...@gmail.com` on 25 Jul 2010 at 11:26

**Comment by GoogleCodeExporter** [comment]
Original comment by `anti...@gmail.com` on 30 Aug 2010 at 10:55
- Changed state: **Verified**
